static/js/controllers/inbox.js

```diff
@@ -202,7 +212,11 @@ require('moment/locales');
         });
       };
 
-      $scope.$on('ContactUpdated', function() {
+      $scope.$on('ContactUpdated', function(contact) {
+        if (!contact || contact._deleted) {
+          // allow the ContactCtrl to handle contact deletion
+          return;
+        }
```

Comment:
This will leave the Facilities filter in an inconsistent state - ie: you'll be able to filter for facilities that no longer exist. As above, because this is in the root you should be able to safely allow the request to finish with timeout:false